can/scripts/viewer.py

```diff
@@ -0,0 +1,602 @@
+#!/usr/bin/python
+# coding: utf-8
+#
+# Copyright (C) 2018 Kristian Sloth Lauszus. All rights reserved.
+#
+# Contact information
+# -------------------
+# Kristian Sloth Lauszus
+# Web      :  http://www.lauszus.com
+# e-mail   :  lauszus@gmail.com
+
+from __future__ import absolute_import, print_function
+
+import argparse
+import can
+import curses
+import os
+import six
+import struct
+import sys
+
+from curses.ascii import ESC as KEY_ESC, SP as KEY_SPACE
+from typing import Dict, List, Tuple, Union
+
+from can import __version__
+
+# CANopen function codes
```

Comment:
Python-can is meant to be protocol agnostic, so if we support CANopen then we will have to prepare to support many different protocols in the future as well. Perhaps it would be a good idea to make it possible to write plugins for decoding messages. There are a lot of proprietary CAN protocols out there and this would allow the most flexibility. We could also link to third party packages from here so it is easy to find them.

Reply:
I agree that the viewer should remain protocol agnostic. Can we aim to support multiple CAN protocols without the user having to change any `python-can` code?

Your separate config file for `--decode` looks close for a naive (no custom protocol code) version? Could add a display name per frame id.